### PR TITLE
Adding internet check to bitbar version plugin

### PR DESCRIPTION
**Pull request by oliviernt**
prevents showing and error diretly when re-connecting to wifi

### CODE REVIEW

**Comment by keithamus** [comment]
Hmm... would it not be easier to just try/catch on fetching the new version, rather than pinging google.com?

**Comment by oliviernt** [comment]
You're right. I missed the easy change :D

File: BitBar/version.12h.rb [hunk 1]
```diff
@@ -16,6 +16,7 @@
 require 'net/http'
 require 'json'
 require 'nokogiri'
+require 'resolv'
```
Comment (keithamus) [inline BitBar/version.12h.rb]: Is resolv needed any more?

File: BitBar/version.12h.rb [hunk 2]
```diff
@@ -26,8 +27,22 @@
...
+  while json_result.nil?
+    begin
+      json_result = JSON.parse(Net::HTTP.get(url))
+    rescue => e
+      sleep_time *= 2
+      if sleep_time < sleep_until
+        sleep sleep_time
+      else
+        throw e
+      end
```
Comment (keithamus) [inline BitBar/version.12h.rb]: I like the exponential backoff, very clever 👍

File: BitBar/version.12h.rb (hunk 1 above)
Comment (oliviernt) [inline BitBar/version.12h.rb]: Yup. Missed that.

**Pull request merged**